src/api/interfaces/frame_system.rs

```diff
@@ -0,0 +1,171 @@
+use ac_primitives::{AccountInfo, ExtrinsicParams, FrameSystemConfig};
+use log::*;
+use serde::de::DeserializeOwned;
+use sp_core::{storage::StorageKey, Pair};
+use sp_runtime::{generic::SignedBlock, traits::GetRuntimeBlockType, MultiSignature};
+
+pub trait GetAccountInformation<AccountId> {
+	type Index;
+	type AccountData;
+
+	fn get_account_info(
+		&self,
+		address: &AccountId,
+	) -> ApiResult<Option<AccountInfo<Self::Index, Self::AccountData>>>;
+
+	fn get_account_data(&self, address: &AccountId) -> ApiResult<Option<Self::AccountData>>;
+}
+
+pub trait GetHeader<Hash> {
+	type Header;
+
+	fn get_finalized_head(&self) -> ApiResult<Option<Hash>>;
+
+	fn get_header(&self, hash: Option<Hash>) -> ApiResult<Option<Self::Header>>;
+}
+
+pub trait GetBlock<Number, Hash> {
+	type Block;
+
+	fn get_block_hash(&self, number: Option<Number>) -> ApiResult<Option<Hash>>;
+
+	fn get_block(&self, hash: Option<Hash>) -> ApiResult<Option<Self::Block>>;
+
+	fn get_block_by_num(&self, number: Option<Number>) -> ApiResult<Option<Self::Block>>;
+
+	/// A signed block is a block with Justification ,i.e., a Grandpa finality proof.
+	/// The interval at which finality proofs are provided is set via the
+	/// the `GrandpaConfig.justification_period` in a node's service.rs.
+	/// The Justification may be None.
+	fn get_signed_block(&self, hash: Option<Hash>) -> ApiResult<Option<SignedBlock<Self::Block>>>;
+
+	fn get_signed_block_by_num(
+		&self,
+		number: Option<Number>,
+	) -> ApiResult<Option<SignedBlock<Self::Block>>>;
+}
+
+impl<Signer, Client, Params, Runtime> GetAccountInformation<Runtime::AccountId>
+	for Api<Signer, Client, Params, Runtime>
+where
+	Signer: Pair,
+	MultiSignature: From<Signer::Signature>,
+	Client: Request,
+	Runtime: FrameSystemConfig,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+{
+	type Index = Runtime::Index;
+	type AccountData = Runtime::AccountData;
+
+	fn get_account_info(
+		&self,
+		address: &Runtime::AccountId,
+	) -> ApiResult<Option<AccountInfo<Self::Index, Self::AccountData>>> {
+		let storagekey: StorageKey = self.metadata().storage_map_key::<Runtime::AccountId>(
+			"System",
+			"Account",
+			address.clone(),
+		)?;
+
+		info!("storage key is: 0x{}", hex::encode(&storagekey));
+		self.get_storage_by_key_hash(storagekey, None)
+	}
+
+	fn get_account_data(
+		&self,
+		address: &Runtime::AccountId,
+	) -> ApiResult<Option<Runtime::AccountData>> {
+		self.get_account_info(address).map(|info| info.map(|i| i.data))
+	}
+}
+
+impl<Signer, Client, Params, Runtime> GetHeader<Runtime::Hash>
+	for Api<Signer, Client, Params, Runtime>
+where
+	Client: Request,
+	Runtime: FrameSystemConfig,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime::Header: DeserializeOwned,
+	Runtime::Hash: FromHexString,
+{
+	type Header = Runtime::Header;
+
+	fn get_finalized_head(&self) -> ApiResult<Option<Runtime::Hash>> {
+		let finalized_block_hash =
+			self.client().request("chain_getFinalizedHead", rpc_params![])?;
```

Comment:
this is obviously not from the `frame-system-pallet`

Reply:
Moved to chain.rs